### ISSUE TITLE
大佬拜托了，这个问题卡了三天了，帮帮可怜的孩子吧，好人一生平安

### ISSUE THREAD

**Issue by Jerry-155** [ali-vilab/ACE_plus#113]
@yaosheng216  大佬能帮我看看是哪里出问题了嘛，为什么我自己训练的模型出的图是模糊的呢？十分感谢

这是我训练的subject模型生的图![Image](https://github.com/user-attachments/assets/ff3705f1-fd69-4ed8-8380-541c21d47dde)

这是官方subject模型生的图![Image](https://github.com/user-attachments/assets/33f7cc9f-22fc-4d44-b869-8e8809be465d)

这是我的配置文件
ENV:
  BACKEND: nccl
  SEED: 1999
SOLVER:

  NAME: FormalACEPlusSolver

  MAX_STEPS: 12000

  USE_AMP: True

  DTYPE: bfloat16
  ENABLE_GRADSCALER: False

  USE_FAIRSCALE: False
  USE_ORIG_PARAMS: True
  USE_FSDP: True  
default: False
  LOAD_MODEL_ONLY: False

  RESUME_FROM:

  WORK_DIR: ./examples/exp_example/

  LOG_FILE: std_log.txt

  LOG_TRAIN_NUM: 16

  FSDP_REDUCE_DTYPE: bfloat16

  FSDP_BUFFER_DTYPE: bfloat16

  FSDP_SHARD_MODULES:
    - MODULE: 'model.model'
      FSDP_GROUP: [ 'single_blocks', 'double_blocks']
    - MODULE: 'cond_stage_model.t5_model.hf_module.encoder'
      FSDP_GROUP: [ 'block' ] #
  SAVE_MODULES: [ 'model'] #
  TRAIN_MODULES: ['model']

  FILE_SYSTEM:
    - NAME: HuggingfaceFs
      TEMP_DIR: ./cache
    - NAME: ModelscopeFs
      TEMP_DIR: ./cache
  #
  MODEL:
    NAME: LatentDiffusionACEPlus
    PARAMETERIZATION: rf
    TIMESTEPS: 1000
    GUIDE_SCALE: 1.0
    PRETRAINED_MODEL:
    IGNORE_KEYS: [ ]
    USE_EMA: False
    EVAL_EMA: False
    SIZE_FACTOR: 8
    DIFFUSION:
      NAME: DiffusionFluxRF
      PREDICTION_TYPE: raw
      NOISE_NORM: True
      # NOISE_SCHEDULER DESCRIPTION:  TYPE:  default: ''
      NOISE_SCHEDULER:
        NAME: FlowMatchFluxShiftScheduler
        SHIFT: False
        PRE_T_SAMPLE: True
        PRE_T_SAMPLE_FOLD: 1
        SIGMOID_SCALE: 1
        BASE_SHIFT: 0.5
        MAX_SHIFT: 1.15
      SAMPLER_SCHEDULER:
        NAME: FlowMatchFluxShiftScheduler
        SHIFT: True
        PRE_T_SAMPLE: False
        SIGMOID_SCALE: 1
        BASE_SHIFT: 0.5
        MAX_SHIFT: 1.15

    #
    DIFFUSION_MODEL:
      # NAME DESCRIPTION:  TYPE:  default: 'Flux'
      NAME: FluxMRACEPlus
      PRETRAINED_MODEL: ${FLUX_FILL_PATH}/flux1-fill-dev.safetensors
      # IN_CHANNELS DESCRIPTION: model's input channels. TYPE: int default: 64
      IN_CHANNELS: 384
      # OUT_CHANNELS DESCRIPTION: model's input channels. TYPE: int default: 64
      OUT_CHANNELS: 64
      # HIDDEN_SIZE DESCRIPTION: model's hidden size. TYPE: int default: 1024
      HIDDEN_SIZE: 3072
      REDUX_DIM: 1152
      # NUM_HEADS DESCRIPTION: number of heads in the transformer. TYPE: int default: 16
      NUM_HEADS: 24
      # AXES_DIM DESCRIPTION: dimensions of the axes of the positional encoding. TYPE: list default: [16, 56, 56]
      AXES_DIM: [ 16, 56, 56 ]
      # THETA DESCRIPTION: theta for positional encoding. TYPE: int default: 10000
      THETA: 10000
      # VEC_IN_DIM DESCRIPTION: dimension of the vector input. TYPE: int default: 768
      VEC_IN_DIM: 768
      # GUIDANCE_EMBED DESCRIPTION: whether to use guidance embedding. TYPE: bool default: False
      GUIDANCE_EMBED: True
      # CONTEXT_IN_DIM DESCRIPTION: dimension of the context input. TYPE: int default: 4096
      CONTEXT_IN_DIM: 4096
      # MLP_RATIO DESCRIPTION: ratio of mlp hidden size to hidden size. TYPE: float default: 4.0
      MLP_RATIO: 4.0
      # QKV_BIAS DESCRIPTION: whether to use bias in qkv projection. TYPE: bool default: True
      QKV_BIAS: True
      # DEPTH DESCRIPTION: number of transformer blocks. TYPE: int default: 19
      DEPTH: 19
      # DEPTH_SINGLE_BLOCKS DESCRIPTION: number of transformer blocks in the single stream block. TYPE: int default: 38
      DEPTH_SINGLE_BLOCKS: 38
      # ATTN_BACKEND：setting 'flash_attn' to use flash_attn2, if the version of pytorch > 2.4.0, using 'pytorch' to use pytorch's implementation
      ATTN_BACKEND: flash_attn
      # USE_GRAD_CHECKPOINT: setting gc to true can decrease the memory usage.
      SWIFT_LORA_MODEL:  "models/comfyui_subject_lora16.safetensors"
      USE_GRAD_CHECKPOINT: True

    #
    FIRST_STAGE_MODEL:
      NAME: AutoencoderKLFlux
      EMBED_DIM: 16
      PRETRAINED_MODEL: ${FLUX_FILL_PATH}/ae.safetensors
      IGNORE_KEYS: [ ]
      BATCH_SIZE: 8
      USE_CONV: False
      SCALE_FACTOR: 0.3611
      SHIFT_FACTOR: 0.1159
      #
      ENCODER:
        NAME: Encoder
        CH: 128
        OUT_CH: 3
        NUM_RES_BLOCKS: 2
        IN_CHANNELS: 3
        ATTN_RESOLUTIONS: [ ]
        CH_MULT: [ 1, 2, 4, 4 ]
        Z_CHANNELS: 16
        DOUBLE_Z: True
        DROPOUT: 0.0
        RESAMP_WITH_CONV: True
      #
      DECODER:
        NAME: Decoder
        CH: 128
        OUT_CH: 3
        NUM_RES_BLOCKS: 2
        IN_CHANNELS: 3
        ATTN_RESOLUTIONS: [ ]
        CH_MULT: [ 1, 2, 4, 4 ]
        Z_CHANNELS: 16
        DROPOUT: 0.0
        RESAMP_WITH_CONV: True
        GIVE_PRE_END: False
        TANH_OUT: False
    #
    COND_STAGE_MODEL:
      # NAME DESCRIPTION:  TYPE:  default: 'T5PlusClipFluxEmbedder'
      NAME: T5ACEPlusClipFluxEmbedder
      # T5_MODEL DESCRIPTION:  TYPE:  default: ''
      T5_MODEL:
        # NAME DESCRIPTION:  TYPE:  default: 'HFEmbedder'
        NAME: ACEHFEmbedder
        # HF_MODEL_CLS DESCRIPTION: huggingface cls in transfomer TYPE: NoneType default: None
        HF_MODEL_CLS: T5EncoderModel
        # MODEL_PATH DESCRIPTION: model folder path TYPE: NoneType default: None
        MODEL_PATH: ${FLUX_FILL_PATH}/text_encoder_2/
        # HF_TOKENIZER_CLS DESCRIPTION: huggingface cls in transfomer TYPE: NoneType default: None
        HF_TOKENIZER_CLS: T5Tokenizer
        # TOKENIZER_PATH DESCRIPTION: tokenizer folder path TYPE: NoneType default: None
        TOKENIZER_PATH: ${FLUX_FILL_PATH}/tokenizer_2/
        ADDED_IDENTIFIER: [ '<img>','{image}', '{caption}', '{mask}', '{ref_image}', '{image1}', '{image2}', '{image3}', '{image4}', '{image5}', '{image6}', '{image7}', '{image8}', '{image9}' ]
       
        MAX_LENGTH: 512
  
        OUTPUT_KEY: last_hidden_state
 
        D_TYPE: bfloat16

        BATCH_INFER: False
        CLEAN: whitespace

      CLIP_MODEL:

        NAME: ACEHFEmbedder

        HF_MODEL_CLS: CLIPTextModel

        MODEL_PATH: ${FLUX_FILL_PATH}/text_encoder/

        HF_TOKENIZER_CLS: CLIPTokenizer

        TOKENIZER_PATH: ${FLUX_FILL_PATH}/tokenizer/
     
        MAX_LENGTH: 77

        OUTPUT_KEY: pooler_output
     
        D_TYPE: bfloat16

        BATCH_INFER: True
        CLEAN: whitespace
  TUNER:
    # THE LORA PARAMETERS
    - NAME: SwiftLoRA
      R: 32
      LORA_ALPHA: 32
      LORA_DROPOUT: 0.0
      BIAS: "none"
      TARGET_MODULES: "(model.double_blocks.*(.qkv|.img_mlp.0|.img_mlp.2|.txt_mlp.0|.txt_mlp.2|.proj|.img_mod.lin|.txt_mod.lin))|(model.single_blocks.*(.linear1|.linear2|.modulation.lin))$"
      PRETRAINED_MODEL: "models/comfyui_subject_lora16.safetensors"
  SAMPLE_ARGS:
    SAMPLE_STEPS: 28
    SAMPLER: flow_euler
    SEED: 42
    IMAGE_SIZE: [ 1024, 1024 ]
    GUIDE_SCALE: 30

  LR_SCHEDULER:
    NAME: StepAnnealingLR
    WARMUP_STEPS: 0
    TOTAL_STEPS: 100000
    DECAY_MODE: 'cosine'
  #
  OPTIMIZER:
    NAME: AdamW
    LEARNING_RATE: 1e-3
    BETAS: [ 0.9, 0.999 ]
    EPS: 1e-6
    WEIGHT_DECAY: 1e-2
    AMSGRAD: False
  #
  TRAIN_DATA:
    NAME: ACEPlusDataset
    MODE: train
    DATA_LIST: data/train.csv
    DELIMITER: "#;#"
    # input_image, input_mask, input_reference_image, target_image, instruction, task_type
    FIELDS: ["edit_image", "edit_mask", "ref_image", "target_image", "prompt", "data_type"]
    PATH_PREFIX: ""
    EDIT_TYPE_LIST: []
    MAX_SEQ_LEN: 2048
    D: 16
    PIN_MEMORY: True
    BATCH_SIZE: 1
    NUM_WORKERS: 4
    SAMPLER:
      NAME: LoopSampler

  EVAL_DATA:
    NAME: ACEPlusDataset
    MODE: eval
    DATA_LIST: data/eval.csv
    DELIMITER: "#;#"
    # input_image, input_mask, input_reference_image, target_image, instruction, task_type
    FIELDS: [ "edit_image", "edit_mask", "ref_image", "target_image", "prompt", "data_type" ]
    PATH_PREFIX: ""
    EDIT_TYPE_LIST: [ ]
    MAX_SEQ_LEN: 2048
    D: 16
    PIN_MEMORY: True
    BATCH_SIZE: 1
    NUM_WORKERS: 4

  TRAIN_HOOKS:
    - NAME: ACEBackwardHook
      GRADIENT_CLIP: 1.0
      PRIORITY: 10
    - NAME: LogHook
      LOG_INTERVAL: 20
    - NAME: ACECheckpointHook
      INTERVAL: 1000
      PRIORITY: 200
      DISABLE_SNAPSHOT: True
    - NAME: ProbeDataHook
      PROB_INTERVAL: 1500
      PRIORITY: 0
    - NAME: TensorboardLogHook
      LOG_INTERVAL: 50
  EVAL_HOOKS:
    - NAME: ProbeDataHook
      PROB_INTERVAL: 1500
      PRIORITY: 0

**Comment by yaosheng216** [comment]
@Jerry-155 可以提供你的GPU型号以及显存信息吗，根据现有提供的配置信息，MAX_SEQ_LEN值设置的确实有些小，这会导致训练的LoRA图片存在清晰度较差的问题，可以尝试增加GPU的显存而扩大MAX_SEQ_LEN，以便训练出效果更好的LoRA

**Comment by Jerry-155** [comment]
> [@Jerry-155](https://github.com/Jerry-155) 可以提供你的GPU型号以及显存信息吗，根据现有提供的配置信息，MAX_SEQ_LEN值设置的确实有些小，这会导致训练的LoRA图片存在清晰度较差的问题，可以尝试增加GPU的显存而扩大MAX_SEQ_LEN，以便训练出效果更好的LoRA

@yaosheng216 这次图片使用的是gpu是A100 40g，我之前尝试过用H20 80G显存的显卡把MAX_SEQ_LEN调到了4096训练的结果依然是模糊的，我想知道你们官方subject等系列lora模型，训练的时候MAX_SEQ_LEN的值是多少，或者说这个值最低要开到多少才能保证图片的清晰度，十分感谢您的回复

**Comment by yaosheng216** [comment]
![Image](https://github.com/user-attachments/assets/cea4dec9-4ed3-4301-953e-fe550cc41844)
@Jerry-155 MAX_SEQ_LEN这个参数限制的是合并到一起的图的seq，单图4096，如果是两张图合并到一起则应是8192（4096*2）

**Comment by Jerry-155** [comment]
@yaosheng216 所以MAX_SEQ_LEN的值要调整到8192甚至更高才能保证图片的清晰度嘛

**Comment by yaosheng216** [comment]
> [@yaosheng216](https://github.com/yaosheng216) 所以MAX_SEQ_LEN的值要调整到8192甚至更高才能保证图片的清晰度嘛

参考生成或参考编辑，有一个参考图的情况下MAX_SEQ_LEN应为 2*4096（1个参考图，1个生成图）。另外，我们修复了一些训练适配的问题，你可以重新拉取下最新的代码去进行训练

**Comment by gggarlic** [comment]
> > [@yaosheng216](https://github.com/yaosheng216) 所以MAX_SEQ_LEN的值要调整到8192甚至更高才能保证图片的清晰度嘛
> 
> 参考生成或参考编辑，有一个参考图的情况下MAX_SEQ_LEN应为 2*4096（1个参考图，1个生成图）。另外，我们修复了一些训练适配的问题，你可以重新拉取下最新的代码去进行训练

对应的config里IMAGE_SIZE: [ 1024, 1024 ]要调成[4096,4096]吗
